### PR TITLE
feat: enable nursery lints without findings

### PR DESCRIPTION
**Pull request by jtraglia**
This is quite verbose, but I like the idea of enabling all of the nursery (experimental) lints which did not have any findings. I'm hoping this prevents future issues while the codebase grows. I've also listed the ones which we allow (disable) so we know what to fix in the future. Also, fix `nonstandard_macro_braces` issue; it was very minor & not worth its own PR.

These were copy/pasted from this page:

* https://rust-lang.github.io/rust-clippy/master/index.html#/?groups=nursery

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
any objections @DaniPopes ?

**Review by onbjerg — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -87,16 +87,64 @@ rust.unreachable_pub = "warn"
...
+suspicious_operation_groupings = "deny"
+trailing_empty_array = "deny"
+trait_duplication_in_bounds = "deny"
+transmute_undefined_repr = "deny"
+trivial_regex = "deny"
+tuple_array_conversions = "deny"
+uninhabited_references = "deny"
+unused_peekable = "deny"
+unused_rounding = "deny"
+useless_let_if_seq = "deny"
```
Comment (DaniPopes) [inline Cargo.toml:124]: Not a big fan of denying lints, and we deny them in clippy CI anyway.

Comment (jtraglia) [reply Cargo.toml:124]: What do you mean? I don't think I understand.

Comment (DaniPopes) [reply Cargo.toml:124]: Use "warn" instead, as we `--deny warnings` in CI to catch them.

Comment (jtraglia) [reply Cargo.toml]: Oh gotcha, yeah that's fair. Changed them to warn.

**Pull request merged**